core/src/main/java/com/graphhopper/storage/CHGraphImpl.java

```diff
@@ -387,8 +394,13 @@ void initStorage() {
 
     @Override
     public CHGraph create(long bytes) {
-        nodesCH.create(bytes);
-        shortcuts.create(bytes);
+        // creating the DataAccess here does not make much sense, because nodesCH has to have a certain size that we
+        // only know in prepareForContraction(). also the size of shortcuts is unknown and if anything it should be
+        // approximated when the number of base graph edges can be determined -> also in prepareForContraction()
+        // Anyway, we have to create the DataAccess here, because otherwise we get an error when calling flush() and
+        // loadExisting() later, see #2384
+        nodesCH.create(0);
+        shortcuts.create(0);
```

Comment:
Maybe we should in general remove the size argument of this create method? And only create the basics or already one segment in DataAccess.

Reply:
Yes, good question. `x.create(bytes)` would already be possible by using `x.create(); x.ensureCapacity(bytes)`?

Reply:
Yes, indeed. This should be identical.

Reply:
-> https://github.com/graphhopper/graphhopper/issues/2386